src/pages/InputDemo/InputDemo.jsx

```diff
@@ -1,61 +1,153 @@
 import React from 'react';
+import {
+  schema, selectOptions, radioOptionsCricket, radioOptionsFootball,
+} from '../../config/constant';
 import TextField from '../../components/TextField/TextField';
 import SelectField from '../../components/SelectField/selectField';
-import RadioGroup from '../../components/RadioGroup/radioGroup';
-import { selectOptions, radioOptionsCricket, radioOptionsFootball } from '../../config/constant';
+import { RadioGroup } from '../../components/RadioGroup/index';
+import { Button } from '../../components/Button/index';
 
 
 class InputDemo extends React.Component {
   constructor(props) {
     super(props);
-
     this.state = {
       name: '',
       sport: '',
       position: '',
+      error: {
+      },
+      isvalid: false,
+      touched: {
+        name: false,
+        sport: false,
+        position: false,
+      },
     };
   }
 
   onChangeTextField = (event) => {
-    this.setState({ name: event.target.value });
+    this.setValueAndError({ name: event.target.value }, () => { this.setError(); });
   }
 
   onChangeSelectOptions = (event) => {
```

Comment:
use single handler instead of onChangeSelectOption, onChangeRadioOption, onChangeTextField. It should be handled with a single event handler